app/pages/admin/project-status.spec.js

```diff
@@ -33,68 +33,73 @@ const workflows = [
-  let onChangeWorkflowLevelSpy;
-  let handleDialogCancelSpy;
-  let handleDialogSuccessSpy;
-  let handleToggleSpy;
-  let event;
-
+  let onChangeWorkflowLevelStub;
+  let wrapper;
 
   before(() => {
-    onChangeWorkflowLevelSpy = sinon.stub(ProjectStatus.prototype, 'onChangeWorkflowLevel');
-    handleDialogCancelSpy = sinon.stub(ProjectStatus.prototype, 'handleDialogCancel');
-    handleDialogSuccessSpy = sinon.stub(ProjectStatus.prototype, 'handleDialogSuccess');
-    handleToggleSpy = sinon.stub(ProjectStatus.prototype, 'handleToggle');
+    onChangeWorkflowLevelStub = sinon.stub(ProjectStatus.prototype, 'onChangeWorkflowLevel');
+    handleDialogCancelStub = sinon.stub(ProjectStatus.prototype, 'handleDialogCancel');
+    handleDialogSuccessStub = sinon.stub(ProjectStatus.prototype, 'handleDialogSuccess');
 
     wrapper = shallow(<ProjectStatus />);
-
-    wrapper.setState({
-      project,
-      workflows
-    });
   });
 
   it('renders without crashing', () => {
     assert.equal(wrapper, wrapper);
   });
 
-  describe('custom components', () => {
-    it('LoadingIndicator renders if project is not in state', () => {
-      wrapper.setState({ project: null });
+  describe('when no project is in component state', () => {
+    it('renders Loading Indicator component', () => {
       loadingIndicator = wrapper.find('LoadingIndicator');
       assert.equal(loadingIndicator.length, 1);
     });
+  });
 
-    it('LoadingIndicator does not render if project is in state', () => {
+  describe('when project is in component state', () => {
+    before(() => {
       wrapper.setState({ project });
+    });
+
+    it('does not render the LoadingIndicator component', () => {
       loadingIndicator = wrapper.find('LoadingIndicator');
       assert.equal(loadingIndicator.length, 0);
     });
 
-    it('ProjectIcon renders', () => {
+    it('renders a ProjectIcon component', () => {
+      wrapper.setState({ project });
       const projectIconComponent = wrapper.find('ProjectIcon');
       assert.equal(projectIconComponent.length, 1);
     });
 
-    it('RedirectToggle renders', () => {
+    it('renders a RedirectToggle component', () => {
       const redirectToggleComponent = wrapper.find('RedirectToggle');
       assert.equal(redirectToggleComponent.length, 1);
     });
 
-    it('ExperimentalFeatures renders', () => {
+    it('renders a ExperimentalFeatures component', () => {
       const experimentalFeaturesComponent = wrapper.find('ExperimentalFeatures');
       assert.equal(experimentalFeaturesComponent.length, 1);
     });
 
-    it('VersionList renders', () => {
+    it('renders a VersionList component', () => {
       const versionListComponent = wrapper.find('VersionList');
       assert.equal(versionListComponent.length, 1);
     });
+
+    it('renders a no workflows found message when no workflow is in component state', () => {
+      const noWorkflowsMessage = wrapper.find('div[children="No workflows found"]');
```

Comment:
I was able to target this element using a solution [suggested here](https://github.com/airbnb/enzyme/issues/270), but, as one person comments, this seems tightly coupled to the implementation - would it be better to move this text to a Translate component, and then test whether what is rendered matches what is in the Translate component's `content` prop? ([something more like this](https://github.com/zooniverse/Panoptes-Front-End/blob/ba4b0ca140dccbb325264ff66b0b6af5ddb67ef8/app/pages/security-policy.spec.js#L24))

Reply:
We don't need translations here. I'm of the opinion that finding specific DOM elements for a unit test will always be coupled with the implementation. It's unavoidable, however, I would avoid using a prop that is internal to the React API as well as using a selector that looks like a CSS selector when it's not. Here's how I'd write this test and the DOM traversal:
```
      const noWorkflowsMessage = wrapper.find('.project-status__section').at(1).children().find('div');
      assert.equal(noWorkflowsMessage.text(), 'No workflows found');
```
It also better matches the description of the test, which is that it renders a specific message.